packages/blade/src/components/StepGroup/_decisions/decisions.md

```diff
@@ -0,0 +1,219 @@
+| isSelected   | Selected state of item                                                                            | boolean                                                            | undefined                             |
+| href         | Anchor tag's href value. Turns StepItem into interactive item and render it as anchor             | string                                                             | undefined                             |
+| onClick      | StepItem's click handler. Turns StepItem into interactive item and render it as button tag        | () => void                                                         | undefined                             |
+
+### StepItemIcon
+
+| **Props** | **Description** | **Type**                                                                        | **Default Value** |
+| --------- | --------------- | ------------------------------------------------------------------------------- | ----------------- |
+| icon      | icon component  | IconComponent                                                                   |                   |
+| color     | color of icon   | 'positive' \| 'negative' \| 'neutral' \| 'notice' \| 'information' \| 'primary' | 'neutral'         |
+
+### StepItemIndicator
+
+| **Props** | **Description**    | **Type**                                                                        | **Default Value** |
+| --------- | ------------------ | ------------------------------------------------------------------------------- | ----------------- |
+| color     | color of indicator | 'positive' \| 'negative' \| 'neutral' \| 'notice' \| 'information' \| 'primary' | 'neutral'         |
+
+## Usage
+
+### Static StepGroup
+
+Static StepGroup does not have any hover, focus, and active states. It can't be selected or clicked. It is for static steps only.
+
+<img width="200px" src="./2024-04-12-14-12-50.png" alt=""/>
+
+```jsx
+<StepGroup>
+  <StepItem
+    leading={<StepItemIndicator color="positive" />}
+    title="Disputes Raised"
+    timestamp="Thu, 11th Oct23 | 12:00pm"
+    stepProgress="full"
+  />
+  <StepItem
+    leading={<StepItemIndicator color="notice" />}
+    title="Needs Response"
+    timestamp="Respond latest by Tue, 23rd Oct'24 | 12:00pm"
+    description="Description"
+    stepProgress="start"
+  >
+    <Button variant="secondary">Submit Documents</Button>
+  </StepItem>
+  <StepItem
+    leading={<StepItemIndicator color="neutral" />}
+    title="Decision from Bank"
+    trailing={<Badge color="neutral">Pending</Badge>}
+  />
+</StepGroup>
+```
+
+### Interactive StepGroup
+
+Interactive StepGroup can be used to add clickable items in StepGroup. When StepItem has `onClick` or `href` prop, it becomes interactive.
+
+<img width="200px" src="./2024-04-12-15-16-25.png" alt="" />
+
+```jsx
+const InteractiveStepGroup = () => {
+  const [selectedIndex, setSelectedIndex] = React.useState(-1);
```

Comment:
How to handle interactive + nested step group?

Reply:
this index is anyway consumer-side only so they can either just give 1, 2, 3, 4, indexes in sequential manner indepdent of nesting and handle

Or they can keep state like `useState('1-2')` to represent second item in first StepGroup. Either way eventually they just have to give `isSelected` to selected item somehow